wappsto/connection/network_classes/value.py

```diff
@@ -471,13 +394,66 @@ def update(self, data_value, timestamp=None):
             state.timestamp = timestamp
         else:
             state.timestamp = self.get_now()
-        self.last_update_of_report = state.timestamp
 
         return self.parent.parent.conn.send_state(
             state,
             data_value=data_value,
         )
 
+    def check_delta_and_period(self, data_value):
+        """
+        Check if delta and period allows data to be sent.
+
+        Check if value has delta or period, if it has then if it passes
+        checks then True is returned, otherwise False is returned.
+
+        Args:
+            data_value: the new value.
+
+        Returns:
+            True/False indicating the result of operation.
+
+        """
+        if (self.delta is not None and self.__is_number_type()):
+            # delta should work
+            data_value = float(data_value)
+            if (self.last_update_of_control is None or abs(data_value - self.last_update_of_control) >= self.delta):
+                # delta exeeded
+                self.last_update_of_control = data_value
+                if self.period is not None:
+                    # timer should be reset if period exists
+                    self.__set_timer()
+                return True
+            else:
+                # delta not exeeded
+                return self.check_period(False)
+        return self.check_period(True)
+
+    def check_period(self, return_value):
+        """
+        Check if period allows data to be sent.
+
+        Check if value has period, if it has then if it passes
+        checks then True is returned, otherwise False is returned.
+
+        Args:
+            return_value: default return value.
+
+        Returns:
+            True/False indicating the result of operation.
+
+        """
+        if self.period is not None:
+            # period should work
+            if threading.current_thread() == self.timer or not self.timer.is_alive():
```

Comment:
Is the main.py sending updates in the timer thread or how do you expect this check to work?